### ISSUE TITLE
Change Network parameters in Gluon API

### ISSUE THREAD

**Issue by edmBernard** [apache/incubator-mxnet#7531]
For the example, let say I want to have a dropout probability that change over iteration.
In NDArray API I can do it like that (from gluon tutoriel): 
```python
def net(X, drop_prob=0.0):
    h1_linear = nd.dot(X, W1) + b1
    h1 = relu(h1_linear)
    h1 = dropout(h1, drop_prob)
    h2_linear = nd.dot(h1, W2) + b2
    h2 = relu(h2_linear)
    h2 = dropout(h2, drop_prob)
    yhat_linear = nd.dot(h2, W3) + b3
    return yhat_linear
```
and in the training loop I can do that : 
```python
for e in range(epochs):
    for i, (data, label) in enumerate(train_data):
        data = data.as_in_context(ctx).reshape((-1,784))
        label = label.as_in_context(ctx)
        label_one_hot = nd.one_hot(label, 10)
        with autograd.record():
            output = net(data, drop_prob=.5**e)    # <-- here I can change dropout probability
            loss = softmax_cross_entropy(output, label_one_hot)
        loss.backward()
        SGD(params, learning_rate)
```

How can we do that with the gluon interface ? : 
```python
net = gluon.nn.Sequential()
with net.name_scope():
    net.add(gluon.nn.Dense(num_hidden, activation="relu"))
    net.add(gluon.nn.Dropout(.5))
    net.add(gluon.nn.Dense(num_hidden, activation="relu"))
    net.add(gluon.nn.Dropout(.5))
    net.add(gluon.nn.Dense(num_outputs))
```
In training loop: 
```python
for e in range(epochs):
    for i, (data, label) in enumerate(train_data):
        data = data.as_in_context(ctx).reshape((-1, 784))
        label = label.as_in_context(ctx)
        with autograd.record():
            output = net(data)                #  <-- how can I change dropout factor
            loss = softmax_cross_entropy(output, label)
            loss.backward()
        trainer.step(data.shape[0])
```

**Comment by szha** [comment]
You should be able to do that by changing the dropout rate for the layer. However, this won't work with symbolic or hybridized mode. For imperative mode, you can do:
```python
net[1]._rate = e%10 * 0.1
```

**Comment by edmBernard** [comment]
@szha thanks for your answer, my example was not enough accurate to describe what I wanted.
I find another solution: 
```python
    class Model(gluon.Block):
        def __init__(self, num_outputs, **kwargs):
            super(Model, self).__init__(**kwargs)
            # use name_scope to give child Blocks appropriate names.
            # It also allows sharing Parameters between Blocks recursively.
            with self.name_scope():
                self.dense0 = gluon.nn.Dense(128)
                self.dense1 = gluon.nn.Dense(64)
                self.dense3_1 = gluon.nn.Dense(num_outputs)
                self.dense3_2 = gluon.nn.Dense(num_outputs)

        def forward(self, x, use_branch1):
            root = nd.relu(self.dense1(nd.relu(self.dense0(x))))
            if use_branch1:
                out = self.dense3_1(root)
            else:
                out = self.dense3_2(root)
            return out
```
I put my full example [here](https://github.com/edmBernard/mxnet_example_shared_weight/blob/master/demo_with_gluon_inside_param.py) the purpose was to resolve this https://github.com/apache/incubator-mxnet/issues/7530
